### DISCUSSION TITLE
Potential bug when adding and getting links

### DISCUSSION THREAD

**Discussion by chapmanwilliam** [pymupdf/PyMuPDF#3142] [Looking for help] [answered]
Two potential bugs:

1) When adding pages dynamically, the page.get_links() function fails to return the links on the last page _unless_ you add one further page before calling the function. Only then is page.first_link not None.
2) Only on a mac, even then, when you add a link it seems to add page=7&zoom=100,0,0 to the filespec variable (rather than just page=7) which causes linkDest to fail because linkDest parses the page numbrer by reference to everything after the first five characters. I have done a workaround at lines 6791 of the module 'fitz__init__py

              if len(ftab) == 2:
                    if ftab[1].startswith("page="):
                        self.kind = LINK_GOTOR
                        self.fileSpec = ftab[0]
                        print(ftab[1][5:])
                        x=ftab[1][5:].split('&')[0]
                        self.page = int(x) - 1

Conceivably, this might have something to do with the different path types on Mac and PC because I didn't use to have this problem until I started using the Path library to handle paths.

**Comment by chapmanwilliam** [comment]
Jorji, if I want to create a branch on your code, is that possible?

**Reply by JorjMcKie** [reply]
No - not that. But you can fork and then make a branch there. The submit a PR using that branch with your changes ... if that is your intention.

**Comment by JorjMcKie** [comment, marked as answer]
Otherwise, when you add an annotation or link, etc. you must reload the page before you can see the results.
This reloading makes sure that all updates actually have been applied and a fresh page representation has been fetched into memory: `page = doc.reload-page(page)` ... please ensure to do it exactly like this.
Alternatively set the `page` object to a different page first and then load it again with the desired page number.
